src/generic/infobar.cpp

```diff
@@ -355,9 +355,10 @@ size_t wxInfoBarGeneric::GetButtonCount() const
     {
         const wxSizerItem * const item = node->GetData();
 
-        // if we reached the spacer separating the buttons from the text
-        // break the for-loop.
-        if ( item->IsSpacer() )
+        // If we reached the spacer separating the buttons from the text
+        // break the for-loop. Same for if we get to the sizer holding
+        // the icon and text.
+        if ( item->IsSpacer() || item->IsSizer() )
```

Comment:
I wonder if it would be worth extracting this test into a helper function instead of repeating it several times. The advantage would be that we could use some descriptive name for it, e.g. `IsButtonItem()` (here the test for it would be reversed).

What do you think?